closes #59

ChuckPierce opened the pull request:
This is my stripe integration implementation.  Please let me know what you think about it when you can.  Doesn't have to be right away.  I'm having problems with clearing all of the data on the form after its sent but I guess we would want to redirect anyway to a confirmation page I suppose.  

Hope you guys are having a good day.

meliis commented on server/api/order/order.controller.js:
```diff
@@ -22,10 +23,28 @@ exports.show = function(req, res) {
+      card: req.body.billing.stripeToken,
+      description: "email@email.com"
+    }, function(err,charge) {
+          if(err && err.type === 'StripeCardError') {
```
little confused why you have two conditions with the same result

ChuckPierce replied:
ah yes the StripeCardError is if the card is declined...the other is just an err in general.  I found that if the card was accepted but the amount on the charge was not a number it would still get past that first conditional.  I threw in the second one so I could catch the not a number result.  I think you would send two different messages back to the frontend so I should change that.

meliis replied:
i think the error message _would_ be different, so there's no need for two different error handlers.

meliis commented:
so there's no way to move the stripe stuff out of the express route? :(

ChuckPierce commented:
well I'm not sure...I mean I could create like a separate stripe creation document and then export it in.  I could then just call a function or something.  But I think you would want it in the route where you're creating an order.

meliis commented:
going with convention i've changed the names of ProductFactory and OrderFactory to Product and Order, by the way
i was hoping you could put the stripe things in like some sort of mongoose middleware??  idk.  you definitely could put the order.close() there, but if that's all you can shift i don't know if it's worth it.

ChuckPierce commented:
Okay I will tackle all of this soon

ChuckPierce commented:
So I just pushed up some changes...I changed the methods on the order so that it will check for a guest order and then apply the correct status accordingly.  Also, I moved the stripe stuff into a static method so that can be called before an Order is even created.  This implementation still works after testing it. Let me know if it's all good :-)

meliis commented:
this is awesome!  dang you even wrote more tests too.  you'll probably still have to explain stuff to us once we all get together again but it looks good to me.

ChuckPierce commented:
Okay is it all right to merge?

meliis commented:
i think so, i feel like katie probably wants to look at it first though

ChuckPierce commented:
Okay I can wait until she sees it

katiepeters commented:
Gave it a once-over. You've been busy, Chuck! I think it's ok to merge.

ChuckPierce commented:
Okay awesome! :+1:
